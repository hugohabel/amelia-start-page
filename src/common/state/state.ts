/* eslint-disable no-case-declarations */
/* eslint-disable no-param-reassign */
// External Dependencies
import produce, { enableES5 } from 'immer';

// Internal Dependencies
import {
  AppAction,
  AppState,
  CountdownWidget,
  Event,
  Location,
  WeatherWidget,
} from '../types/AppState';

if (typeof Proxy === 'undefined') {
  enableES5(); // Shouldn't be necessary if the app runs just on Chrome.
}

export const initialAppState: AppState = {
  widgets: [
    {
      type: 'countdown',
      enabled: true,
      data: {
        events: [
          {
            id: 'ba4d-13dbf795d755',
            name: 'Christmas',
            date: '2021-12-25T05:00:00.000Z',
            background: 'christmas-bg.jpg',
          },
        ],
      },
    },
  ],
  sidebar: {
    isOpen: false,
    activeMenu: null,
    activeView: null,
  },
};

// @TO-DO - Extract the logic here to an external utils function and add tests to it.
export const rootReducer = produce((draft: AppState, action: AppAction) => {
  switch (action.type) {
    case 'addNewWidget':
      draft.widgets.push(action.widget);
      break;
    case 'toggleSidebar':
      draft.sidebar.isOpen = !draft.sidebar.isOpen;
      break;
    case 'setActiveMenu':
      draft.sidebar.activeMenu = action.activeMenu;
      break;
    case 'setActiveView':
      draft.sidebar.activeView = action.activeView;
      break;
    case 'addNewCountdownEvent':
      const countdownWidget: CountdownWidget = draft.widgets.find((widget) => {
        return widget.type === 'countdown';
      });

      if (countdownWidget) {
        countdownWidget.data.events.push(action.event);
      }

      break;
    case 'removeWidgetItem':
      const widgetsList = draft.widgets.find((widget) => {
        return widget.type === action.widgetType;
      });

      if (action.widgetType === 'countdown') {
        const countdownWidgetEventToRemove = widgetsList?.data.events.findIndex((event: Event) => {
          return event.id === action.uuid;
        });

        if (countdownWidgetEventToRemove !== -1) {
          widgetsList?.data.events.splice(countdownWidgetEventToRemove, 1);
        }
      } else if (action.widgetType === 'weather') {
        const weatherWidgetEventToRemove = widgetsList?.data.locations.findIndex(
          (location: Location) => {
            return location.id === action.uuid;
          }
        );

        if (weatherWidgetEventToRemove !== -1) {
          widgetsList?.data.locations.splice(weatherWidgetEventToRemove, 1);
        }
      }

      break;
    case 'addNewWeatherLocation': {
      const weatherWidget: WeatherWidget = draft.widgets.find((widget) => {
        return widget.type === 'weather';
      });

      if (weatherWidget) {
        weatherWidget.data.locations.push(action.location);
      } else {
        const newWeatherWidget: WeatherWidget = {
          type: 'weather',
          enabled: true,
          data: {
            locations: [
              {
                ...action.location,
              },
            ],
          },
        };

        draft.widgets.push(newWeatherWidget);
      }
      break;
    }
    default:
      break;
  }

  return draft;
});
