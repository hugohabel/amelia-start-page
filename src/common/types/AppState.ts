import { CountdownFormat } from './amelia';

// App State
export interface IAppState {
  widgets: any[];
  sidebar: ISidebar;
}

// Sidebar
export interface ISidebar {
  isOpen: boolean;
  activeMenu: TActiveMenu;
  activeView: TActiveView;
}

// Countdown Widget
export interface ICountdownWidget {
  type: 'countdown';
  enabled: boolean;
  data: {
    events: IEvent[];
  };
}

export interface IEvent {
  id: string;
  name: string;
  date: string;
  format: CountdownFormat;
}
// End Countdown Widget

// Weather Widget
export interface IWeatherWidget {
  type: 'weather';
  enabled: boolean;
  data: {
    locations: Location[];
  };
}

export interface ILocation {
  id: string;
  lat: string;
  lng: string;
}
// End Weather Widget

// Auxiliary Types
export type TActiveMenu = 'add' | 'about' | 'config' | null;
export type TActiveView = 'add_countdown' | 'add_weather' | null;

// App State Actions
type TAction =
  | { type: 'addNewWidget'; widget: ICountdownWidget | IWeatherWidget }
  | { type: 'toggleSidebar' }
  | { type: 'setActiveMenu'; activeMenu: TActiveMenu }
  | { type: 'setActiveView'; activeView: TActiveView }
  | { type: 'addNewCountdownEvent'; event: IEvent }
  | { type: 'addNewWeatherLocation'; location: Location }
  | { type: 'removeWidgetItem'; uuid: string; widgetType: string };

export type TAppAction = TAction;
export type TAppDispatch = (action: TAppAction) => void;
