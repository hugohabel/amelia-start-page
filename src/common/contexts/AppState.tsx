// External Dependencies
import { createContext, ReactNode, useContext, useReducer } from 'react';
import { debounce } from 'lodash';

// Internal Dependencies
import { LOCAL_STORAGE_STATE_KEY } from '../constants';
import { TAppAction, TAppDispatch, IAppState } from '../types/AppState';
import { SafeLocalStorage } from '../utils/localStorage/localStorage';
import { initialAppState, rootReducer } from '../state/state';

// Interfaces + Types
interface IAppStateProviderProps {
  children: ReactNode;
  initialStateOverride?: IAppState;
}

// Utility Functions

/**
 * Retrieves state from local storage or initializes with default state.
 *
 * @param defaultState AppState   Default app state.
 * @returns AppState  State (can be an existing one, or a default one)
 */
function getStateFromLocalStorageOrInitialize(defaultState: IAppState) {
  const stateFromLocalStorage = SafeLocalStorage.getInstance().getItem(LOCAL_STORAGE_STATE_KEY);
  if (stateFromLocalStorage) {
    return JSON.parse(stateFromLocalStorage) as IAppState;
  }
  return defaultState;
}

/**
 * Persists app state in memory or in local storage
 *
 * @param state AppState  State to be persisted.
 */
function persistState(state: IAppState) {
  SafeLocalStorage.getInstance().setItem(LOCAL_STORAGE_STATE_KEY, JSON.stringify(state));
}

const debouncePersistState = debounce(persistState, 500, {
  leading: false,
  trailing: true,
  maxWait: 1000,
});

function reducerWrapper(state: IAppState, action: TAppAction) {
  const nextState = rootReducer(state, action);
  debouncePersistState(nextState);
  return nextState;
}

// Create Contexts
const AppStateContext = createContext<IAppState>(initialAppState);
const AppDispatchContext = createContext<TAppDispatch>(() =>
  // eslint-disable-next-line no-console
  console.warn('Calling default dispatch is a no-op!')
);

/**
 * App State Provider
 *
 * @param param AppStateProviderProps Props to initialize the app state provider.
 * @returns JSX App Dispatch and App State Contexts.
 */
export function AppStateProvider({ children, initialStateOverride }: IAppStateProviderProps) {
  const hydratedInitialAppState = {
    ...initialAppState,
  };

  const [state, dispatch] = useReducer(
    reducerWrapper,
    initialStateOverride ?? hydratedInitialAppState,
    getStateFromLocalStorageOrInitialize
  );

  return (
    <AppDispatchContext.Provider value={dispatch}>
      <AppStateContext.Provider value={state}>{children}</AppStateContext.Provider>
    </AppDispatchContext.Provider>
  );
}

AppStateProvider.defaultProps = {
  initialStateOverride: undefined,
};

export const useAppState = () => useContext(AppStateContext);
export const useAppDispatch = () => useContext(AppDispatchContext);
