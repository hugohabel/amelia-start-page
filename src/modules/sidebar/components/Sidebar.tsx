// Internal Dependencies
import { MENUBAR, ACTIONS } from '../../../common/constants';
import { useAppState, useAppDispatch } from '../../../common/contexts/AppState';
import { WidgetsList } from '../../widgetsList/components/WidgetsList';
import CountdownFormAddView from '../../countdownFormAddView/components/CountdownFormAddView';
import { CloseIcon } from '../../../common/components/icons/CloseIcon';

// Types
import { ISidebar } from '../../../common/types/AppState';

// Styles
import styles from './Sidebar.module.css';

// Utility Functions

function viewHandler(sidebar: ISidebar): number {
  let activeView = 0;

  if (sidebar.activeMenu === null) {
    activeView = 0;
  } else {
    activeView += MENUBAR[sidebar.activeMenu.toUpperCase()];
  }

  if (!sidebar.activeView || sidebar.activeView === null) {
    activeView += 0;
  } else {
    activeView += ACTIONS[sidebar.activeView.toUpperCase()];
  }

  return activeView;
}

function mapStateToView(activeView: number) {
  switch (activeView) {
    case 0:
      return null;
    case 1:
      return <WidgetsList />;
    case 11:
      return <CountdownFormAddView />;
    default:
      return null;
  }
}

function Sidebar() {
  const { sidebar } = useAppState();
  const dispatch = useAppDispatch();
  const sidebarClasses = sidebar.isOpen
    ? `${styles.sidebar} ${styles.isSidebarOpen}`
    : `${styles.sidebar}`;
  const sidebarInnerContainerClasses = sidebar.isOpen
    ? `${styles.sidebarInnerContainer} ${styles.sidebarInnerContainerOpen}`
    : `${styles.sidebarInnerContainer}`;

  // Handle Functions
  function handleCloseIconClick() {
    /* istanbul ignore else */
    if (sidebar.isOpen) {
      dispatch({
        type: 'toggleSidebar',
      });
      dispatch({
        type: 'setActiveMenu',
        activeMenu: null,
      });
      dispatch({
        type: 'setActiveView',
        activeView: null,
      });
    }
  }

  return (
    <div className={sidebarClasses}>
      {/* Close Icon */}
      <div
        data-testid="close-sidebar"
        className={styles.sidebarCloseIcon}
        onClick={() => handleCloseIconClick()}>
        <CloseIcon />
      </div>
      {/* End Close Icon */}

      {/* Main Sidebar View */}
      <div className={sidebarInnerContainerClasses}>{mapStateToView(viewHandler(sidebar))}</div>
      {/* End Main Sidebar View */}
    </div>
  );
}

export default Sidebar;
