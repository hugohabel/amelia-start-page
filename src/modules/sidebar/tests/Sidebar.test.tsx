// External Dependencies
import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';

// Internal Dependencies
import { AppStateProvider, useAppDispatch } from '../../../common/contexts/AppState';
import Sidebar from '../components/Sidebar';
import { initialAppState } from '../../../common/state/state';

// Mocked Functions
jest.mock('../../../common/contexts/AppState', () => ({
  ...(jest.requireActual('../../../common/contexts/AppState') as object),
  useAppDispatch: jest.fn(),
}));
let mockAppDispatch: ReturnType<typeof useAppDispatch>;

// Mocked Values
const DEFAULTSTATE = {
  ...initialAppState
};

describe('Sidebar', () => {
  it('matches snapshot - with sidebar closed', async () => {
    const component = render(
      <AppStateProvider initialStateOverride={initialAppState}>
        <Sidebar />
      </AppStateProvider>
    );

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('matches snapshot - with sidebar opened', async () => {
    initialAppState.sidebar.isOpen = true;

    const component = render(
      <AppStateProvider initialStateOverride={initialAppState}>
        <Sidebar />
      </AppStateProvider>
    );

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('matches snapshot - with sidebar opened - WidgetsList', async () => {
    initialAppState.sidebar.isOpen = true;
    initialAppState.sidebar.activeMenu = 'add';
    initialAppState.sidebar.activeView = null;

    const component = render(
      <AppStateProvider initialStateOverride={initialAppState}>
        <Sidebar />
      </AppStateProvider>
    );

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('matches snapshot - with sidebar opened - CountdownFormAddView', async () => {
    initialAppState.sidebar.isOpen = true;
    initialAppState.sidebar.activeMenu = 'add';
    initialAppState.sidebar.activeView = 'add_countdown';

    const component = render(
      <AppStateProvider initialStateOverride={initialAppState}>
        <Sidebar />
      </AppStateProvider>
    );

    expect(component.asFragment()).toMatchSnapshot();
  });

  describe('Sidebar - Handlers', () => {

    beforeEach(() => {
      mockAppDispatch = jest.fn();
      (useAppDispatch as jest.Mock).mockReturnValue(mockAppDispatch);
    });

    it('triggers state updates when clicking on the close icon', async () => {
      DEFAULTSTATE.sidebar.isOpen = true;

      const component = render(
        <AppStateProvider initialStateOverride={DEFAULTSTATE}>
          <Sidebar />
        </AppStateProvider>
      );

      // Get Go Back Element
      const closeElement = await component.findByTestId('close-sidebar');
      expect(closeElement).toBeInTheDocument();

      // Trigger Click
      fireEvent.click(closeElement);

      await waitFor(() => {
        expect(mockAppDispatch).toHaveBeenCalledTimes(3);
      });
    });
  });
});