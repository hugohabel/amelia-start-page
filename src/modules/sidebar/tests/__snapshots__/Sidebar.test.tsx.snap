// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar matches snapshot - with sidebar closed 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
  >
    <div
      class="sidebarCloseIcon"
      data-testid="close-sidebar"
    >
      <svg
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="sidebarInnerContainer"
    />
  </div>
</DocumentFragment>
`;

exports[`Sidebar matches snapshot - with sidebar opened - CountdownFormAddView 1`] = `
<DocumentFragment>
  <div
    class="sidebar isSidebarOpen"
  >
    <div
      class="sidebarCloseIcon"
      data-testid="close-sidebar"
    >
      <svg
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="sidebarInnerContainer sidebarInnerContainerOpen"
    >
      <div
        class="backContainer"
        data-testid="go-back"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
        <span>
          Go Back
        </span>
      </div>
      <h2
        class="title"
      >
        New countdown
      </h2>
      <p
        class="subtitle"
      >
        Add the details for your countdown widget
      </p>
      <form
        autocomplete="off"
      >
        <div
          class="inputContainer"
        >
          <div
            class="emojiPlaceholder"
            data-testid="event-emoji"
          >
            ðŸ’¡
          </div>
          <input
            class="eventNameInput"
            id="eventName"
            name="eventName"
            placeholder="Event name"
            type="text"
            value=""
          />
        </div>
        <input
          id="eventDate"
          name="eventDate"
          placeholder="Event date"
          type="date"
          value=""
        />
        <h4
          class="subtitle"
        >
          How do you want to display the countdown?
        </h4>
        <div
          class="MuiFormGroup-root radioGroup"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
              style="color: rgb(149, 128, 255);"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-4"
                  name="eventFormat"
                  type="radio"
                  value="days+hours"
                />
                <div
                  class="PrivateRadioButtonIcon-root-5 PrivateRadioButtonIcon-checked-7"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Days and Hours
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              style="color: rgb(149, 128, 255);"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-4"
                  name="eventFormat"
                  type="radio"
                  value="days"
                />
                <div
                  class="PrivateRadioButtonIcon-root-5"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Days
            </span>
          </label>
        </div>
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sidebar matches snapshot - with sidebar opened - WidgetsList 1`] = `
<DocumentFragment>
  <div
    class="sidebar isSidebarOpen"
  >
    <div
      class="sidebarCloseIcon"
      data-testid="close-sidebar"
    >
      <svg
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="sidebarInnerContainer sidebarInnerContainerOpen"
    >
      <div
        class="widgetsList"
      >
        <h2
          class="widgetsListTitle"
        >
          New widget
        </h2>
        <p
          class="widgetsListSubtitle"
        >
          Select a widget from the list below
        </p>
        <div
          class="widgetsListItem"
          data-testid="countdown"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 2v6h.01L6 8.01 10 12l-4 4 .01.01H6V22h12v-5.99h-.01L18 16l-4-4 4-3.99-.01-.01H18V2H6zm10 14.5V20H8v-3.5l4-4 4 4zm-4-5l-4-4V4h8v3.5l-4 4z"
            />
          </svg>
          <span>
            Countdown
          </span>
        </div>
        <div
          class="widgetsListItem widgetsListItemUpcoming"
          data-testid="weather"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.76 4.84l-1.8-1.79-1.41 1.41 1.79 1.79zM1 10.5h3v2H1zM11 .55h2V3.5h-2zm8.04 2.495l1.408 1.407-1.79 1.79-1.407-1.408zm-1.8 15.115l1.79 1.8 1.41-1.41-1.8-1.79zM20 10.5h3v2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm0 10c-2.21 0-4-1.79-4-4s1.79-4 4-4 4 1.79 4 4-1.79 4-4 4zm-1 4h2v2.95h-2zm-7.45-.96l1.41 1.41 1.79-1.8-1.41-1.41z"
            />
          </svg>
          <span>
            Weather
          </span>
          <span
            class="widgetsListItemUpcomingTitle"
          >
            Upcoming!
          </span>
        </div>
        <div
          class="widgetsListItemHidden"
          data-testid="timezone"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm.5-13H11v6l5.2 3.2.8-1.3-4.5-2.7V7z"
            />
          </svg>
          <span>
            Timezone Converter
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sidebar matches snapshot - with sidebar opened 1`] = `
<DocumentFragment>
  <div
    class="sidebar isSidebarOpen"
  >
    <div
      class="sidebarCloseIcon"
      data-testid="close-sidebar"
    >
      <svg
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="sidebarInnerContainer sidebarInnerContainerOpen"
    />
  </div>
</DocumentFragment>
`;
